src/ZODB/_compat.py

```diff
@@ -61,14 +72,58 @@ def loads(s):
     FILESTORAGE_MAGIC = b"FS30"
 
 
-# XXX: consistent spelling of inst_persistent_id/persistent_id?
-#      e.g. StorageTestBase and probably elsewhere
+def PersistentPickler(persistent_id, *args, **kwargs):
+    """
+    Returns a :class:`Pickler` that will use the given ``persistent_id``
+    to get persistent IDs. The remainder of the arguments are passed to the
+    Pickler itself.
+
+    This covers the differences between Python 2 and 3 and PyPy/zodbpickle.
+    """
+    p = Pickler(*args, **kwargs)
+    if sys.version_info[0] < 3:
+        p.inst_persistent_id = persistent_id
+        # PyPy uses a python implementation of cPickle in both Python 2
+        # and Python 3. We can't really detect inst_persistent_id as its
+        # a magic attribute that's not readable, but it doesn't hurt to
+        # simply always assign to persistent_id also
+        p.persistent_id = persistent_id
+    else:
+        p.persistent_id = persistent_id
```

Comment:
Will do. I just moved existing code here from `ZODB.serialize`, but it's code that I originally wrote :)